lib/execute/src/execute.rs

```diff
@@ -154,26 +158,73 @@ pub fn execute(
         }
     }
 
-    let code_buf =
-        &compilation.functions[module
-                                   .defined_func_index(start_index)
-                                   .expect("imported start functions not supported yet")];
-
     // Collect all memory base addresses and Vec.
     let mut mem_base_addrs = instance
         .memories
         .iter_mut()
         .map(LinearMemory::base_addr)
         .collect::<Vec<_>>();
-    let vmctx = make_vmctx(instance, &mut mem_base_addrs);
+
+    Ok(make_vmctx(instance, &mut mem_base_addrs))
+}
+
+/// Jumps to the code region of memory and execute the start function of the module.
+pub fn start(
+    module: &Module,
+    compilation: &Compilation,
+    context: &mut Context,
+) -> Result<(), String> {
+    if context.initialized {
+        return Err(String::from("the start function was already called"));
+    }
+
+    let start_index = module
+        .start_func
+        .ok_or_else(|| String::from("No start function defined, aborting execution"))?;
```

Comment:
If there's no start function defined, we should just return here, without reporting an error.